### PR TITLE
Errors with syntax loop

### PR DESCRIPTION
**Pull request by MiuNice**
like: with (col1, col2) as (subquery) as ..., it enters an infinite loop.             
return exception

### CODE REVIEW

**Comment by macbre** [comment]
@MiuNice - thanks for the PR! Can you please add a test case (or two)?

**Comment by MiuNice** [comment]
ok later

**Review by macbre — APPROVED** [review]

File: test/test_with_statements.py [hunk 1]
```diff
@@ -493,3 +495,35 @@ def test_identifier_syntax():
 
     assert parser.tables == ["hits"]
     assert parser.columns == ["EventDate", "EventTime", "ts_upper_bound"]
+
+
+def test_as_was_preceded_by_with_query():
```
Comment (macbre) [inline test/test_with_statements.py:500]: @MiuNice, CI complains about the code formatting here. Can you please run `black` here?

**Comment by MiuNice** [comment]
Sorry for submitting the wrong code due to my oversight

**Comment by MiuNice** [comment]
I apologize for submitting the error code due to my negligence

**Pull request closed without merging**